packages/api-core/src/tests/ms.test.js

```diff
@@ -137,5 +147,137 @@ describe('AvMicroservice', () => {
       const testExpected = '/api/test';
       expect(ms.getUrl(testConfig)).toBe(testExpected);
     });
+
+    test('should return full URL when host is specified', () => {
+      const testUrl = 'https://apps.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        host: 'apps.availity.com',
+      };
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return cloud URL when host is specified', () => {
+      const testUrl = 'https://digital.awp.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        host: 'digital.awp.availity.com',
+      };
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return apps when location is prod cloud', () => {
+      const testUrl = 'https://apps.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        window: {
+          location: {
+            hostname: 'digital.awp.availity.com',
+            pathname: '/cdn/prd/spaces/index.html',
+          },
+        },
+      };
+
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return test-apps when location is tst cloud', () => {
+      const testUrl = 'https://test-apps.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        window: {
+          location: {
+            hostname: 'digital.awn.availity.com',
+            pathname: '/cdn/tst/spaces/index.html',
+          },
+        },
+      };
+
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return qa-apps when location is stg cloud', () => {
+      const testUrl = 'https://qa-apps.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        window: {
+          location: {
+            hostname: 'digital.awn.availity.com',
+            pathname: '/cdn/stg/spaces/index.html',
+          },
+        },
+      };
+
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return t01-apps when location is t01 cloud', () => {
+      const testUrl = 'https://t01-apps.availity.com/api/v1/test';
+      const testConfig = {
+        api: true,
+        path: '/api/',
+        version: '/v1/',
+        name: '/test',
+        window: {
+          location: {
+            hostname: 'digital.awn.availity.com',
+            pathname: '/cdn/t01/spaces/index.html',
+          },
+        },
+      };
+
+      expect(ms.getUrl(testConfig)).toBe(testUrl);
+    });
+
+    test('should return relative URL when location host is prod cloud but path is non-prod', () => {
```

Comment:
I'm not 100% following this unit test. I was expecting by the code that if we are sending a request to `/api/v1` we are hitting our on prem A1 but the app is hosted in the cloud so it would need the fully qualified URL?